nilearn/reporting/tests/test_html_report.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+from nibabel import Nifti1Image
+from nilearn import input_data
+
+
+def _check_html(html_view):
+    """ Check the presence of some expected code in the html viewer
+    """
+    assert "Parameters" in str(html_view)
+    assert "data:image/png;base64," in str(html_view)
+
+
+def test_3d_reports():
+    # Dummy 3D data
+    data = np.zeros((9, 9, 9))
+    data[3:-3, 3:-3, 3:-3] = 10
+    data_img_3d = Nifti1Image(data, np.eye(4))
+    # check 3d datasets
+    mask = input_data.NiftiMasker()
+    mask.fit(data_img_3d)
+    html = mask.generate_report()
+    _check_html(html)
+
+
```

Comment:
could we have a test with a masker fitted on no images (mask image provided in init)?

Reply:
I had previously added a test with a mask image provided to init, but it looks like I was also supplying images to `fit`. I've added another test in https://github.com/nilearn/nilearn/pull/2019/commits/de43ec8a90bf4e50d06299dcb5a87be36b92718e with a mask image supplied to `init` and no images passed to `fit` !